apps/coscrad-frontend/src/components/higher-order-components/global-search.tsx

```diff
@@ -0,0 +1,62 @@
+import { FormFieldType, ResourceCompositeIdentifier, ResourceType } from '@coscrad/api-interfaces';
+import { Box, MenuItem, Select } from '@mui/material';
+import { useState } from 'react';
+import { DynamicSelect } from '../dynamic-forms/dynamic-form-elements/dynamic-select';
+import { fullViewCategorizablePresenterFactory } from '../resources/factories/full-view-categorizable-presenter-factory';
+import { AggregateDetailContainer } from './aggregate-detail-container';
+
+interface GlobalSearchProps {
+    onNewSelection?: (resourceCompositeIdentifier: ResourceCompositeIdentifier) => void;
+}
+
+export const GlobalSearch = ({ onNewSelection }: GlobalSearchProps) => {
+    const [selectedResourceType, setSelectedResourceType] = useState<ResourceType>(null);
+
+    const [selectedId, setSelectedId] = useState<string>(null);
+
+    return (
+        <Box>
+            <Select
```

Comment:
It occurs to me that we could find a way to inject the index presenter and use it as a select, while leveraging its glorious filtering capabilities.